File: 프로그래머스/lv1/12921. 소수 찾기/소수 찾기.py
import math;
def solution(n):
    answer = 0

    for i in range(2,n+1):
        flag = True
        for j in range(2,int(i**0.5)+1):
            if i%j == 0 :
                flag = False
                break
        if flag:
            answer += 1
    return answer

File: 프로그래머스/lv1/12921. 소수 찾기/소수 찾기.java
class Solution {
    public int solution(int n) {
       int [] arr = new int [n+1];
		
		for(int a=2; a<=n; a++) {
			arr[a] = a;
		}
		
		for(int i=2; i<=n; i++) {
			if(arr[i] == 0) {
				continue;
			}
			for(int j=i+i; j<=n; j+=i) {
				arr[j]=0;
			}
		}
		
		int answer = 0;
		for(int i=2; i<=n; i++) {
			if(arr[i]!=0) {
				answer ++;
			}
		}
        
        return answer;
    }
}